chirp/wxui/query_sources.py

```diff
@@ -188,6 +189,70 @@ def do_query(self):
         self.send_end()
 
 
+class RRCALOGINQueryThread(QueryThread):
+    global rrloggedin
+    rrloggedin = False
+
+    def do_query(self):
+        global rrloggedin
+        if rrloggedin:
+            rrloggedin = False
+            self.finalRRquery()
+        else:
+            self.do_loginRR()
+
+    def do_loginRR(self):
+        global clist, provinces, default_county, rrloggedin
```

Comment:
I guess the goal is to have these all be global so that even after we've run one query, we remain in "logged in" state and don't need to re-list provinces and such?

Reply:
Ah. That was the goal, but in the end I abandoned that quest. It's annoying to have to re-login but it's probably the better way to do it. I'm still a bit on the fence about it, but I think re-logging in is the way to go. Open to hearing opinions though!

Reply:
No, it's not actually retaining a "session" just retaining confirmation that the credentials were valid at the time. It re-authenticates with RR when the county is submitted. If the credentials become invalid in the interim (ie, the premium subscription ends) then it would fail despite reportedly being logged in.

Reply:
The vernacular "logged in" is probably not actually true when it's displayed on the status message...

Reply:
Ah, okay cool. So the reason for the login step is that you can't pull the provinces, states, or whatever until you're authed?

Reply:
Irritatingly, yes.